src/common/data_type.rs

```diff
@@ -0,0 +1,511 @@
+                PythonType::Str,
+                SqlType::VARCHAR,
+            )),
+            DataType::LargeUtf8 => Ok(DataTypeMap::new(
+                DataType::LargeUtf8,
+                PythonType::Str,
+                SqlType::VARCHAR,
+            )),
+            DataType::List(_) => Err(py_datafusion_err(DataFusionError::NotImplemented(format!(
+                "{:?}",
+                arrow_type
+            )))),
+            DataType::FixedSizeList(_, _) => Err(py_datafusion_err(
+                DataFusionError::NotImplemented(format!("{:?}", arrow_type)),
+            )),
+            DataType::LargeList(_) => Err(py_datafusion_err(DataFusionError::NotImplemented(
+                format!("{:?}", arrow_type),
+            ))),
+            DataType::Struct(_) => Err(py_datafusion_err(DataFusionError::NotImplemented(
+                format!("{:?}", arrow_type),
+            ))),
+            DataType::Union(_, _, _) => Err(py_datafusion_err(DataFusionError::NotImplemented(
+                format!("{:?}", arrow_type),
+            ))),
+            DataType::Dictionary(_, _) => Err(py_datafusion_err(DataFusionError::NotImplemented(
+                format!("{:?}", arrow_type),
+            ))),
+            DataType::Decimal128(_, _) => Err(py_datafusion_err(DataFusionError::NotImplemented(
+                format!("{:?}", arrow_type),
+            ))),
+            DataType::Decimal256(_, _) => Err(py_datafusion_err(DataFusionError::NotImplemented(
+                format!("{:?}", arrow_type),
+            ))),
+            DataType::Map(_, _) => Err(py_datafusion_err(DataFusionError::NotImplemented(
+                format!("{:?}", arrow_type),
+            ))),
+            DataType::RunEndEncoded(_, _) => Err(py_datafusion_err(
+                DataFusionError::NotImplemented(format!("{:?}", arrow_type)),
+            )),
+        }
+    }
+}
+
+#[pymethods]
+impl DataTypeMap {
+    #[new]
+    pub fn py_new(arrow_type: PyDataType, python_type: PythonType, sql_type: SqlType) -> Self {
+        DataTypeMap {
+            arrow_type,
+            python_type,
+            sql_type,
+        }
+    }
+
+    #[staticmethod]
+    #[pyo3(name = "arrow")]
+    pub fn py_map_from_arrow_type(arrow_type: &PyDataType) -> PyResult<DataTypeMap> {
+        DataTypeMap::map_from_arrow_type(&arrow_type.data_type)
+    }
+
+    #[staticmethod]
+    #[pyo3(name = "sql")]
+    pub fn map_from_sql_type(sql_type: &SqlType) -> PyResult<DataTypeMap> {
+        match sql_type {
+            SqlType::ANY => Err(py_datafusion_err(DataFusionError::NotImplemented(format!(
+                "{:?}",
+                sql_type
+            )))),
+            SqlType::ARRAY => Err(py_datafusion_err(DataFusionError::NotImplemented(format!(
+                "{:?}",
+                sql_type
+            )))),
+            SqlType::BIGINT => Ok(DataTypeMap::new(
+                DataType::Int64,
+                PythonType::Float,
```

Comment:
```suggestion
                PythonType::Int,
```

Reply:
I should have left a comment on this. This also wasn't what I thought but seems like a BIGINT wouldn't fit in a Python int and that a float should be used? I really don't know I was just going off of this [information](https://learn.microsoft.com/en-us/sql/machine-learning/python/python-libraries-and-data-types?view=sql-server-ver16) what do you think @martin-g

Reply:
I don't have time now to read the Python docs but I noticed that it inconsistent with https://github.com/apache/arrow-datafusion-python/pull/173/files#diff-a109fe9b61a82ff498f40948cfa040798d7d40213c7df7669df526f914b22a04R82-R86 and added this suggestion.